[clarencepohh] iP

clarencepohh opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

clarencepohh committed:
Renamed chatbot to Hachi; Updated all instances of Duke to Hachi. Changed logo to match new name.

clarencepohh committed:
Rename chatbot to Hachi

clarencepohh committed:
Abstracted greeting into its own function [greet()] and changed to call greet() in main()

clarencepohh committed:
Abstract greeting into its own function greet()

clarencepohh committed:
Added spacerInsert function to create a line break with ~.

clarencepohh committed:
Add spacerInsert function to create line breaks

clarencepohh committed:
Add goodbye() function to terminate program

clarencepohh committed:
Added goodbye() function to terminate program.

clarencepohh committed:
Add simple comments

clarencepohh committed:
Added simple comments

clarencepohh committed:
Add documentation as per SE-EDU standards

clarencepohh committed:
Add documentation as per SE-EDU standards

clarencepohh committed:
Merge branch 'master' of https://github.com/clarencepohh/ip

clarencepohh committed:
Add echo functionality

clarencepohh committed:
Add retrieve list and add task functionality

clarencepohh committed:
Add mark and unmark functionality

clarencepohh committed:
Add mark and unmark messages

clarencepohh committed:
Tweak code to comply with coding standards

clarencepohh committed:
Add subclass of Task for Deadline, Event and Todo

To complete implementation to allow user to create todos, events and deadlines. Currently no working implementation for these classes.

clarencepohh committed:
Implement deadline functionality

clarencepohh committed:
Implement event and todo functionality

clarencepohh committed:
Do minor code refactoring

Created Parser and Ui classes in preparation for future abstraction

clarencepohh committed:
Attempt automated text-ui-test

Currently stuck on error when running runtest.bat; Throws an exception after the first line in test.txt is executed: java.util.NoSuchElementException: No line found

nikhil-2101 commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+                case "EVENT":
+                case "DEADLINE":
+                    TaskType currentTask;
+                    if (cleanedInput.startsWith("TODO")) {
```
try your best to avoid having too many nested if/loop statements!

nidhi-nayak replied:
maybe to do this, you can add an "&" in the condition lines to avoid nested statements, but still have your coding logic in tact

nikhil-2101 commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+
+    public static void spacerInsert(String length, boolean hasTab) {
+        String spacer;
+        if (hasTab) {
```
please leave a gap between the initialisation of spacer and the if statement, as per the coding standards

nidhi-nayak replied:
^^ this!! in the future, separating the two blocks of code allows for easier reading

nikhil-2101 commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+            System.out.print("    ");
+        }
+        switch (length) {
+        case "small": // 20 tildes
```
the comments may not be necessary

nikhil-2101 commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+                    break;
+
+                default:
+                    // handle invalid case
```
not necessary to include the comment since the statement printed also indicates the same thing

clarencepohh replied:
I get where you're coming from! The comment was placed there as a placeholder of sorts, so that I can remember to do error handling in the future 😆

nikhil-2101 commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+            }
+        }
+    }
+}
```
Good job overall on having a relatively concise and understandable code! Your documentation also allowed it to be understood easily as well!

nidhi-nayak commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+    private static void printHelpMessage() {
+        spacerInsert("medium", true);
+        System.out.println("You can use the following commands:");
+        System.out.println("    'list' to retrieve your current list of tasks,");
```
just a little note because i was originally doing this too. instead of adding an actual tab you can use the command "\t" to add a tab, and using a lot less spaces (since there is a line limit!) :)

clarencepohh replied:
Thanks for letting me know! I will implement this 👍

nidhi-nayak commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+     * the console for the user to see.
+     *
+     * @param listOfTasks The Task[] array that contains the list of tasks.
+     */
```
good job with your java doc comments!!! :)

nidhi-nayak commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+     */
+
+    public static void addTask(TaskType taskType, Task[] listOfTasks, String line, String cleanInput) {
+        Task toAdd;
```
Again, make sure you add a space between task and the if blocks :)

nidhi-nayak commented on src/main/java/Hachi.java:
```diff
@@ -0,0 +1,266 @@
+     */
+
+    public static void main(String[] args) {
+        spacerInsert("medium", false);
```
I would say for this, make sure you separate your declarations from the rest of the code (keep them on the top and add a one line space between running methods)

nidhi-nayak commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+public class Event extends Task {
+    protected String startDate;
+    protected String endDate;
```
add space! just adds to readability :)

nidhi-nayak reviewed (commented):
Great work overall! Your code was super readable, and the logic was easy to follow. I think all that's left is just some formatting issues, but great work!!! :)

clarencepohh committed:
Fix runtest.bat crashing during runtime

Error was with reinitializing the Scanner in the while loop. Took it out of the loop and the issue was fixed.

clarencepohh committed:
Add exception class and exception for invalid command

clarencepohh committed:
Create relevant exception for error handling

clarencepohh committed:
Merge branch 'branch-Level-5'

clarencepohh committed:
Divide classes into relevant packages

clarencepohh committed:
Update runtest script to reflect new packages

clarencepohh committed:
Merge branch 'branch-A-Packages'

clarencepohh committed:
Fix bug with mark

if mark was called with no following commands an exception would be thrown due to String.substring(-1) being called

clarencepohh committed:
Change taskArray to use ArrayList Collection Class

clarencepohh committed:
Add delete functionality for tasks

clarencepohh committed:
Update text-ui-testing to reflect delete function

clarencepohh committed:
Add save functionality with dummy path name

To be updated to read relative file paths, as well as implement loading functionality

clarencepohh committed:
Add save functionality

Update .gitignore to avoid committing saved tasklist. Update existing methods to utilise file saving capability

clarencepohh committed:
Merge branch 'branch-Level-6'

clarencepohh committed:
Update .gitignore

clarencepohh committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/hachi/Hachi.java

clarencepohh committed:
Fix bug with mark and unmark not locating the last task

clarencepohh committed:
Add MANIFEST.MF

Musfirahe0556596 commented on src/main/java/hachi/data/HachiException.java:
```diff
@@ -0,0 +1,62 @@
+ */
+
+public class HachiException extends Exception{
+    public static final String INVALID_INPUT_MESSAGE = "Not marvelous! I'm not sure what you're saying here...";
+    public static final String EMPTY_TASK_LIST_MESSAGE = "It appears you have no tasks at the moment...";
+    public static final String EMPTY_TASK_DESCRIPTION_MESSAGE = "Where's the task description? " +
+            "Type 'help' without quotes if you need to see the format! ";
+    public static final String MISSING_DEADLINE_BY_DATE_MESSAGE = "Looks like you're missing the complete-by date! Remember: /by <date here>!";
+    public static final String MISSING_EVENT_START_END_DATE_MESSAGE = "Your start or end dates have some formatting issues. It's: /from <start date> /to <end date>";
+    public static final String LIST_OUT_OF_BOUNDS_MESSAGE = "Woah, I'm not so sure if that task exists. You might want to check that task number again.";
+    public static final String CORRUPTED_SAVE_MESSAGE = "The data stored looks corrupted. Getting rid of it.";
```
java coding standard: Line length should be no longer than 120 chars

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+package hachi;
+
+import hachi.data.HachiException;
+import hachi.data.task.*;
```
Suggest to not use wildcard imports! Imported classes should always be listed explicitly

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+
+public class Hachi {
+
+    private static ArrayList<Task> taskArrayList = new ArrayList<>();
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+                String firstWord;
+                int indexOfSpace = cleanedInput.indexOf(" ");
+
+                if (indexOfSpace == -1) { // check for single-word inputs
+                    firstWord = cleanedInput;
+                } else {
+                    firstWord = cleanedInput.substring(0, indexOfSpace);
+                }
```
may want to consider refactoring into a new method to avoid long methods

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+        }
+    }
+
+    private static void save(ArrayList<Task> taskArrayList) throws IOException {
```
for method name, perhaps you may want to follow your loadFile method so you know what you are saving  (so eg. saveFile) :] and remember that plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+        for (Task task : taskArrayList) {
+            if (task != null) {
+                fw.write(task.getSaveFormat() + "\n");
+            } else break;
```
The else loop body should be wrapped by curly brackets irrespective of how many lines there are in the body ! so e.g. 
else {
    break;
}

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+        System.out.println("\tGoodbye! Hope you have a marvelous day.");
+    }
+
+    private static void loadFile(ArrayList<Task> taskArrayList) throws FileNotFoundException, HachiException {
```
good method name! remember that plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+        Scanner s = new Scanner(taskFile);
+
+        while (s.hasNext()) {
+            String[] line = s.nextLine().split(" \\| ");
```
remember that plural form should be used on names representing a collection of objects :]

clarencepohh replied:
Hi, can I check which is this referring to? taskFile -> tasksFile maybe?

Musfirahe0556596 replied:
Hello! its referring to line 256! The 
`String[] line = s.nextLine().split(" \\| ");`

should be something like String[] lines for example or anything else more appropriate :)

Musfirahe0556596 commented on src/main/java/hachi/Hachi.java:
```diff
@@ -0,0 +1,410 @@
+        int taskNumber = getDeleteTaskNumber(cleanedInput);
+        HachiException.checkOutOfBounds(taskNumber);
+
+        Task toDelete = taskArrayList.get(taskNumber - 1);
+        String taskType = toDelete.getTaskType();
+        String statusIcon = toDelete.getStatusIcon();
```
just some coding readability minor issue! (understand the code) but perhaps can name to something like "taskToDelete"

Musfirahe0556596 reviewed (commented):
Code looks generally good 👍  Just some minor coding standard and quality to follow!

clarencepohh committed:
Add manifest.mf to gitignore

clarencepohh committed:
Fix code quality issues

clarencepohh committed:
Fix relative path to allow saving even through JAR

Add unused storage class to prepare for MoreOOP increment

clarencepohh committed:
Code quality fix

clarencepohh committed:
Add task data file (hachidata.txt) to .gitignore

clarencepohh committed:
Refactor Ui methods into separate Ui class

clarencepohh committed:
Stop tracking hachidata.txt

clarencepohh committed:
Remove redundant calls to scanner after refactoring

clarencepohh committed:
Refactor methods into TaskList class

clarencepohh committed:
Refactor methods into Parser and Storage classes

Currently some methods in Hachi still need to be refactored into Parser and Storage respectively

clarencepohh committed:
Complete refactoring into required classes

clarencepohh committed:
Update method to catch empty by date input

clarencepohh committed:
Add Javadocs for Hachi and HachiException classes

clarencepohh committed:
Add functionality to find matching keywords in existing task list

clarencepohh committed:
Update documentation for HachiException class

clarencepohh committed:
Add Javadocs

clarencepohh committed:
Merge pull request #1 from clarencepohh/branch-Level-8

Branch level 8

clarencepohh committed:
Merge branch 'master' of https://github.com/clarencepohh/ip

clarencepohh committed:
Add JavaDocs for TaskList class

clarencepohh committed:
Add JavaDocs for Parser class

clarencepohh committed:
Add JavaDocs for Storage class

clarencepohh committed:
Add JavaDocs to the Ui class

clarencepohh committed:
Merge pull request #2 from clarencepohh/branch-A-JavaDoc

branch-A-JavaDoc

clarencepohh committed:
Update README.md

clarencepohh committed:
Update README.md

clarencepohh committed:
Update README.md

clarencepohh committed:
Update README.md

clarencepohh committed:
Add find in the help command

clarencepohh committed:
Merge branch 'master' of https://github.com/clarencepohh/ip

clarencepohh committed:
Create _config.yml

clarencepohh committed:
Update README.md

clarencepohh committed:
Bug fix for file loading

clarencepohh committed:
Update .gitignore

clarencepohh committed:
Bug fix for new file created not found in next run

clarencepohh committed:
Remove debug line
